ddtrace/internal/telemetry/writer.py

```diff
@@ -123,11 +159,151 @@ def _flush_events_queue(self):
             self._events_queue = []
         return events
 
+    def enable(self):
+        # type: () -> None
+        """
+        Enable the instrumentation telemetry collection service. If the service has already been
+        activated before, this method does nothing. Use ``disable`` to turn off the telemetry collection service.
+        """
+        if self.status == ServiceStatus.RUNNING:
+            return
+
+        self._enabled = True
+        self.start()
+
+        atexit.register(self.stop)
+
+    def _stop_service(self, *args, **kwargs):
+        # type: (...) -> None
+        super(TelemetryBase, self)._stop_service(*args, **kwargs)
+        self.join()
+
+    def _fork_writer(self):
+        # type: () -> None
+        pass
+
+
+class TelemetryMetricsWriter(TelemetryBase):
+    """
+    Submits Telemetry Metrics events to the datadog agent.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        super(TelemetryMetricsWriter, self).__init__(interval=_get_telemetry_metrics_interval_or_default())
+        self._namespace = MetricNamespace()
+
+    def _flush_namespace_metrics(self):
+        # type () -> List[Metric]
+        """Returns a list of all generated metrics and clears the namespace's list"""
+        log.debug("collect Telemetry Metrics")
```

Comment:
Can we add this log line after we flush the metrics queue (and outside of the `with lock` block). We can then add some more useful info to this log message like the number of metrics flushed.